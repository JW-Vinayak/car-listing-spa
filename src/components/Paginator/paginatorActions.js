export const SET_CURRENT_PAGE = "SET_CURRENT_PAGE";

export const setCurrentPage = pageNumber => ({
  type: SET_CURRENT_PAGE,
  payload: { pageNumber }
});
