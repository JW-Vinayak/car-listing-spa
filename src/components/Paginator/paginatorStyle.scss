.pagination {
  .first-link,
  .prev-link,
  .next-link,
  .last-link {
    @include link;
  }
  list-style: none;
  text-align: center;
  & > li {
    margin: $margin-8 $margin-12;
  display: inline-block;

  }
}
