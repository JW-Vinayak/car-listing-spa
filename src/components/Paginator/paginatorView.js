import React from "react";
import PropTypes from "prop-types";


const Paginator = props => {
  let { currentPage, totalPages, setPage } = props;
  console.log('nav props are', props);
  const setCurrentPage = pageNumber => {
    if (pageNumber < 1 || pageNumber > totalPages) {
      return false;
    }
    setPage(pageNumber);
  };

  return (
    <ul className="pagination">
      <li>
        <a href="# " className="first-link" onClick={() => setCurrentPage(1)}>
          First
        </a>
      </li>
      <li>
        <a href="# " className="prev-link" onClick={() => setCurrentPage(currentPage - 1)}>
          Previous
        </a>
      </li>
      <li>
        Page {currentPage} of {totalPages}
      </li>
      <li>
        <a href="# " className="next-link" onClick={() => setCurrentPage(currentPage + 1)}>
          Next
        </a>
      </li>
      <li>
        <a href="# " className="last-link" onClick={() => setCurrentPage(totalPages)}>
          Last
        </a>
      </li>
    </ul>
  );
};

Paginator.propTypes = {
  currentPage: PropTypes.number.isRequired,
  totalPages: PropTypes.number.isRequired,
  setPage: PropTypes.func.isRequired
}

export default Paginator;
