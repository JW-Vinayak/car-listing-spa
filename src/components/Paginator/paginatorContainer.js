import { connect } from "react-redux";
import Pagination from "./paginatorView";
import { setCurrentPage } from "./paginatorActions";

const mapStateToProps = state => ({
  currentPage: state.pagination.currentPage,
  totalPages: state.cars.totalPages
});

const mapDispatchToProps = dispatch => ({
  setPage: pageNumber => dispatch(setCurrentPage(pageNumber))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Pagination);
