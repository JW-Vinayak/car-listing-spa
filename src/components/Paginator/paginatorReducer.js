import { SET_CURRENT_PAGE } from "./paginatorActions";

const initialState = {
  currentPage: 1
};

export default function paginationReducer(state = initialState, action) {
  switch (action.type) {
    case SET_CURRENT_PAGE:
      return {
        ...state,
        currentPage: action.payload.pageNumber
      };
    default:
      return state;
  }
}
