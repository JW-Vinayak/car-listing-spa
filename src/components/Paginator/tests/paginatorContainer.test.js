import React from "react";
import thunk from "redux-thunk";
import Paginator from "../paginatorContainer";
import configureMockStore from "redux-mock-store";
import renderer from "react-test-renderer";

const middlewares = [thunk];

const mockStore = configureMockStore(middlewares);

let store = mockStore({
  pagination: {
    currentPage: 1
  },
  cars: {
    totalPages: 9
  }
});

describe("Paginator component", () => {
  it("shallow renders without crashing", () => {
    const tree = renderer.create(<Paginator store={store} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
