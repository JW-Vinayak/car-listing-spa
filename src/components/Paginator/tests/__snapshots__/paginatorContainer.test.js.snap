// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator component shallow renders without crashing 1`] = `
<ul
  className="pagination"
>
  <li>
    <a
      className="first-link"
      href="# "
      onClick={[Function]}
    >
      First
    </a>
  </li>
  <li>
    <a
      className="prev-link"
      href="# "
      onClick={[Function]}
    >
      Previous
    </a>
  </li>
  <li>
    Page 
    1
     of 
    9
  </li>
  <li>
    <a
      className="next-link"
      href="# "
      onClick={[Function]}
    >
      Next
    </a>
  </li>
  <li>
    <a
      className="last-link"
      href="# "
      onClick={[Function]}
    >
      Last
    </a>
  </li>
</ul>
`;
